File: Python/929_Unique_Email_Addresses.py
class Solution:
    def numUniqueEmails(self, emails):
        """
        :type emails: List[str]
        :rtype: int
        """
        ans = set()
        for email in emails:
            local,domain = email.split('@')
            if '+' in local: local = local[:local.index('+')]
            local = local.replace('.', '')
            ans.add(local+'@'+domain)
        return len(ans)

File: Go/929_unique_email_addresses.go
func numUniqueEmails(emails []string) int {
    
    filteredEmails := make(map[string]struct{})
    
    for _, email := range emails {
        s := strings.Split(email, "@")
        local, domain := s[0], s[1]
        localSplit := strings.Split(local, "+")
        local = localSplit[0]
        
        local = strings.Replace(local, ".", "", -1)
        
        if _, present := filteredEmails[local + "@" + domain]; !present {
            filteredEmails[local + "@" + domain] = struct{}{}
        }
    }
    return len(filteredEmails)
}
